packages/account/src/Sections/Security/ApiToken/api-token.tsx

```diff
@@ -1,4 +1,262 @@
-import 'Components/api-token/api-token.scss';
+import React from 'react';
+import classNames from 'classnames';
+import { Formik, Form, Field, FormikErrors, FieldProps, FormikHelpers } from 'formik';
+import { useApiToken } from '@deriv/api';
+import { ApiToken as TApitoken } from '@deriv/api-types';
+import { Timeline, Input, Button, ThemedScrollbars, Loading } from '@deriv/components';
+import { getPropertyValue } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
+import { Localize, localize } from '@deriv/translations';
+import { TToken } from 'Types';
+import { ApiTokenContext, ApiTokenArticle, ApiTokenCard, ApiTokenTable } from 'Components/api-token';
+import InlineNoteWithIcon from 'Components/inline-note-with-icon';
+import LoadErrorMessage from 'Components/load-error-message';
+import { getApiTokenCardDetails, TOKEN_LIMITS } from 'Constants/api-token-card-details';
+import './api-token.scss';
+
+type AptTokenState = {
+    api_tokens: NonNullable<TToken[]>;
+    error_message: string;
+};
+
+type TApiTokenForm = {
+    token_name: string;
+    read: boolean;
+    trade: boolean;
+    payments: boolean;
+    trading_information: boolean;
+    admin: boolean;
+};
+
+const ApiToken = observer(() => {
+    const { client, ui } = useStore();
+    const { is_switching } = client;
+    const { is_desktop, is_mobile } = ui;
+
+    const { api_token_data, getApiToken, createApiToken, deleteApiToken, isSuccess, isLoading, isError, error } =
+        useApiToken();
+
+    const [state, setState] = React.useReducer(
+        (prev_state: Partial<AptTokenState>, value: Partial<AptTokenState>) => ({
+            ...prev_state,
+            ...value,
+        }),
+        {
+            api_tokens: [],
+            error_message: '',
+        }
+    );
+
+    React.useEffect(() => {
+        /**
+         * Fetch all API tokens
+         */
+        getApiToken();
+    }, [getApiToken]);
+
+    React.useEffect(() => {
+        /**
+         * Update API token list when new token is created or a token is deleted
+         */
+        if (isSuccess) {
+            setState({
+                api_tokens: getPropertyValue(api_token_data, ['tokens']),
+            });
+        } else if (isError) {
+            setState({
+                error_message: getPropertyValue(error, ['message']),
+            });
+        }
+    }, [isSuccess, api_token_data, isError, error]);
+
+    const initial_form: TApiTokenForm = {
+        token_name: '',
+        read: false,
+        trade: false,
+        payments: false,
+        trading_information: false,
+        admin: false,
+    };
+
+    const validateFields = (values: TApiTokenForm) => {
+        const errors: FormikErrors<TApiTokenForm> = {};
+        const token_name = values?.token_name?.trim();
+
+        if (!token_name) {
+            errors.token_name = localize('Please enter a token name.');
+        } else if (!/^[A-Za-z0-9\s_]+$/g.test(token_name)) {
```

Comment:
**NIT:** Can we move this Regex to a Constant file as well. Easier to maintain code

Reply:
@likhith-deriv is it still relevant?